extensions/unit.lua

```diff
@@ -1,34 +1,41 @@
-function WoWUnit:HasBuff(buffname)
+---@return boolean HasDebuff
+---@param bname string buff name
+function WoWUnit:HasBuffByMe(bname)
   local auras = self.Auras
   for _, aura in pairs(auras) do
-    if aura.Name == buffname then
+    if aura.Name == bname and aura.HasCaster and aura.Caster == Me.ToUnit then
       return true
     end
   end
 
   return false
 end
 
-function WoWUnit:HasBuffByMe(name)
+---@return boolean HasDebuff
+---@param dname string debuff name
+function WoWUnit:HasDebuffByMe(dname)
   local auras = self.Auras
   for _, aura in pairs(auras) do
-    if aura.Name == name and aura.HasCaster and aura.Caster == wector.Game.ActivePlayer.ToUnit then
+    if aura.Name == dname and aura.HasCaster and aura.Caster == Me.ToUnit then
       return true
     end
   end
 
   return false
 end
 
-function WoWUnit:HasDebuffByMe(dname)
+---@return WoWAura
+---@param aname string aura name
+function WoWUnit:GetAuraByMe(aname)
   local auras = self.Auras
+
   for _, aura in pairs(auras) do
-    if aura.Name == dname and aura.HasCaster and aura.Caster == wector.Game.ActivePlayer.ToUnit then
-      return true
+    if aura.Name == aname and aura.HasCaster and aura.Caster == Me.ToUnit then
+      return aura
```

Comment:
In the current version (1.1.6) this will lead to a race condition with the garbage collector. If the table `auras` is cleaned while the returned `aura` is still in use it will cause a crash.

A way to copy an aura instance is already implemented for version 1.1.7 but still require more testing.